constantine/ethereum_bls_signatures.nim

```diff
@@ -510,3 +513,62 @@ func batch_verify*[Msg](pubkeys: openArray[PublicKey], messages: openarray[Msg],
   if verified:
     return cttEthBls_Success
   return cttEthBls_VerificationFailure
+
+func alloc_batch_sig_accumulator*(): ptr BatchSigAccumulator {.libPrefix: prefix_ffi.} =
+  ## Allocates using `alloc0` for a `BatchSigAccumulator`
+  ## so that it can remain an incomplete struct in the C header.
+  result = cast[ptr BatchSigAccumulator](alloc0(sizeof(BatchSigAccumulator)))
+
+proc free_batch_sig_accumulator*(p: ptr BatchSigAccumulator) {.libPrefix: prefix_ffi.} =
+  ## Frees previously `alloc`'d into `buf`
+  dealloc p
+
+func init_batch_sig_accumulator*(
+  ctx: var BatchSigAccumulator,
+  #domainSepTag: ptr UncheckedArray[byte],
+  #domainSepTagLen: int,
+  secureRandomBytes: array[32, byte],
+  #accumSepTag: ptr UncheckedArray[byte],
+  #accumSepTagLen: int
+     ) {.libPrefix: prefix_ffi.} =
+  ## Initializes a Batch BLS Signature accumulator context.
+  ##
+  ## This requires cryptographically secure random bytes
+  ## to defend against forged signatures that would not
+  ## verify individually but would verify while aggregated
+  ## https://ethresear.ch/t/fast-verification-of-multiple-bls-signatures/5407/14
+  ##
+  ## An optional accumulator separation tag can be added
+  ## so that from a single source of randomness
+  ## each accumulatpr is seeded with a different state.
+  ## This is useful in multithreaded context.
+
+  # TODO: The domain separation tag and accumSepTag are both hardcoded in constantine
+  # Just drop them in the arguments here and fill here?
```

Comment:
Technically they are hardcoded for _Ethereum_.

But if we implement the parallel batch verification, the accumSepTag will differ.
I think it's better to make that dynamic so we don't break APIs later on.

It's OK to hardcode the domain separation tag as the C name is `ctt_eth_bls_batch_sig_accumulator` and the Nim code is contained in `ehtereum_bls_signatures.nim`